client/client.go

```diff
@@ -558,21 +574,50 @@ func newDefaultRequester(client *Client, opts *Config) (*defaultRequester, error
 		opts = &Config{}
 	}
 
+	// Validate Basic Auth constraints.
+	if (opts.BasicUser != "" && opts.BasicPass == "") ||
+		(opts.BasicPass != "" && opts.BasicUser == "") {
```

Comment:
This is a bit confusing. Also, why impose artificial restrictions on which user/password combinations are allowed?

Reply:
1. Having a user if you supply the password is needed.
2. Having a password if you have a user:

We are not only using it for identifying a user, but we are allowing this user to be admin! Although basic auth does not enforce this at the protocol level, we certainly do not want to allow Admins to have no password :)

All this says is that if you want to use Basic Auth to identity yourself for role based authentication, you must supply both an identity and a password. In this case we include the Basic Auth header, and the server will interpret the request as a Basic Auth request.

If you do not supply anything here, no header is added. This leaves you to access open access points, if you are using HTTPS.

Reply:
Feel free to open this if I missed something.